timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -66,24 +66,21 @@ BDExchangeMove<RealType>::BDExchangeMove(
       d_sample_per_atom_energy_buffer_(batch_size_ * mol_size_ * N), d_atom_idxs_(get_atom_indices(target_mols)),
       d_mol_offsets_(get_mol_offsets(target_mols)), d_log_weights_before_(num_target_mols_),
       d_log_weights_after_(batch_size_ * num_target_mols_), d_lse_max_before_(1), d_lse_exp_sum_before_(1),
-      d_lse_max_after_(batch_size_), d_lse_exp_sum_after_(batch_size_), d_samples_(batch_size_),
+      d_lse_max_after_(batch_size_), d_lse_exp_sum_after_(batch_size_), d_samples_(batch_size_), d_selected_sample_(1),
       d_quaternions_(round_up_even(QUATERNIONS_PER_STEP * num_proposals_per_move_)),
       d_mh_noise_(num_proposals_per_move), d_num_accepted_(1), d_target_mol_atoms_(batch_size_ * mol_size_),
       d_target_mol_offsets_(num_target_mols_ + 1),
       d_intermediate_sample_weights_(batch_size_ * num_intermediates_per_reduce_),
       d_sample_noise_(num_target_mols_ * num_proposals_per_move_),
       d_sampling_intermediate_(num_target_mols_ * batch_size_), d_translations_(translation_buffer_size),
-      d_sample_segments_offsets_(batch_size_ + 1) {
+      d_sample_segments_offsets_(batch_size_ + 1), d_noise_offset_(1), p_noise_offset_(1) {
 
```

Comment:
do we require that `batch_size <= num_proposals_per_move`? if not, then we should test for this

Reply:
We do not, but will enforce that since that is not a useful way to run this, at least it doesn't seem useful to me.

Reply:
Made that a requirement in https://github.com/proteneer/timemachine/pull/1274/commits/c9fe355d1b20b0cf19ae2775d213e0105191bde3